testing/test_impl.hpp

```diff
@@ -14,12 +14,59 @@
     << "Unexpected " name "."
 
 template <typename T>
-std::string t2string(T t) {
-    return std::to_string(t);
+void unit_test::null_buffer_ptr() {
+    //test create, that it's null
+    auto s = buffer_ptr<T>();
+    EXPECT_FALSE(s) << "Null buffer pointer is not null";
+    
+    /* Use Counts cannot be used right now because the shared_ptr compiler 
+     * implementation and the stdlib definition tend to disagree with with 
+     * the actual counts should be for empty shared_ptrs
+     * https://stackoverflow.com/questions/48885252/c-sharedptr-use-count-for-nullptr
+     */
+    // UnexpectedEq(s.use_count(), 1, "reference count");
```

Comment:
It's OK to omit this test; as your comment indicates, there is ambiguity surrounding the use count of an empty `shared_ptr`. I see that you already have a similar test for a non-empty `buffer_ptr`, and that should be good enough.